index.d.ts

```diff
@@ -0,0 +1,208 @@
+  filterHttpHeaders?: boolean;
+  frameworkName?: string;
+  frameworkVersion?: string;
+  hostname?: string;
+  ignoreUrls?: Array<string | RegExp>;
+  ignoreUserAgents?: Array<string | RegExp>;
+  instrument?: boolean;
+  kubernetesNamespace?: string;
+  kubernetesNodeName?: string;
+  kubernetesPodName?: string;
+  kubernetesPodUID?: string;
+  logLevel?: LogLevel;
+  logger?: Logger;
+  metricsInterval?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+  payloadLogFile?: string;
+  secretToken?: string;
+  serverTimeout?: string; // Also support `number`, but as we're removing this functionality soon, there's no need to advertise it
+  serverUrl?: string;
+  serviceName?: string;
+  serviceVersion?: string;
+  sourceLinesErrorAppFrames?: number;
+  sourceLinesErrorLibraryFrames?: number;
+  sourceLinesSpanAppFrames?: number;
+  sourceLinesSpanLibraryFrames?: number;
+  stackTraceLimit?: number;
+  transactionMaxSpans?: number;
+  transactionSampleRate?: number;
+  verifyServerCert?: boolean;
+}
+
+export interface CaptureErrorOptions {
+  request?: IncomingMessage;
+  response?: ServerResponse;
+  timestamp?: number;
+  handled?: boolean;
+  user?: UserObject;
+  tags?: Tags;
+  custom?: object;
+  message?: string;
+}
+
+export interface Tags {
+  [key: string]: TagValue;
+}
+
+export interface UserObject {
+  id?: string | number;
+  username?: string;
+  email?: string;
+}
+
+export interface ParameterizedMessageObject {
+  message: string;
+  params: Array<any>;
+}
+
+export interface Logger {
+  fatal (msg: string, ...args: any[]): void;
+  fatal (obj: {}, msg?: string, ...args: any[]): void;
+  error (msg: string, ...args: any[]): void;
+  error (obj: {}, msg?: string, ...args: any[]): void;
+  warn (msg: string, ...args: any[]): void;
+  warn (obj: {}, msg?: string, ...args: any[]): void;
+  info (msg: string, ...args: any[]): void;
+  info (obj: {}, msg?: string, ...args: any[]): void;
+  debug (msg: string, ...args: any[]): void;
+  debug (obj: {}, msg?: string, ...args: any[]): void;
+  trace (msg: string, ...args: any[]): void;
+  trace (obj: {}, msg?: string, ...args: any[]): void;
+  [propName: string]: any;
+}
+
+export interface TransactionOptions {
+  startTime?: number;
+  childOf?: Transaction | Span | string; // TODO: This technically accepts other values, but we might not want to document these?
+}
+
+export interface SpanOptions {
+  childOf?: Transaction | Span | string // TODO: This technically accepts other values, but we might not want to document these?
+}
+
+export type CaptureBody = 'off' | 'errors' | 'transactions' | 'all';
+export type CaptureErrorLogStackTraces = 'never' | 'messages' | 'always';
+export type LogLevel = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'fatal';
+
+export type CaptureErrorCallback = (err: Error | null, id: string) => void;
+export type FilterFn = (payload: object) => object | Falsy | void;
```

Comment:
I might be wrong, but void should be included in the Falsy definition, isn't it?

Reply:
`void` isn't technically a falsy value in the pure JavaScript sense of the word. It's more of a lack of explicit `return` from a function which is a concept something TypeScript know about right?

Reply:
By reading [this definition](https://www.typescriptlang.org/docs/handbook/basic-types.html#void) it appears that `void` can be either `null` or `undefined`, which are inside your `Falsy` definition.
I'm not sure how works TypeScript in this regard, but as you know, JavaScript has an implicit `return undefined` if nothing is specified.

Reply:
Hmm you might be half right. it seems that `void` is almost like a synonym for `undefined`. Check out this example:

![image](https://user-images.githubusercontent.com/10602/54695835-fc976280-4b2a-11e9-9839-62d3c71f775c.png)

The only valid uses of return here is `return` by it self or `return undefined`. If I type `return null` it complains with the error you see.

However, if I don't declare `void` as a valid return type, I'm forced to add a `return`:

![image](https://user-images.githubusercontent.com/10602/54696187-aaa30c80-4b2b-11e9-8052-6a275f034a22.png)

Reply:
Fixed